base/pkg/entry.jl

```diff
@@ -383,8 +384,9 @@ function update(branch::AbstractString)
                         if isa(err, LibGit2.GitError)
                             warn("Package $pkg cannot be updated. Error: $(err.msg) Class: $(err.class)")
                         else
-                            rethrow(err)
+                            print_with_color(:red, "Package $pkg cannot be updated. Error:$err")
                         end
+                        push!(defered_errors, CapturedException(err, catch_backtrace()))
```

Comment:
I'd move this line right after ``catch err``, because `catch_backtrace()` might be overwritten by calling other code (not sure if that is the case here.)